nemo/collections/asr/parts/submodules/rnnt_greedy_decoding.py

```diff
@@ -607,6 +615,289 @@ def forward(
 
         return (packed_result,)
 
+    def _greedy_decode_cuda_optimized_lstm(
+        self,
+        x: torch.Tensor,
+        out_len: torch.Tensor,
+        device: torch.device,
+        max_symbols_per_hyp_factor: float = 0.5,
+        partial_hypotheses: Optional[List[rnnt_utils.Hypothesis]] = None,
+    ):
+        if partial_hypotheses is not None:
+            raise NotImplementedError("`partial_hypotheses` support is not supported")
+
+        with torch.inference_mode():
```

Comment:
Preferred to do just torch.no_grad() unless inferece mode is really needed - the modules are already frozen so they are effectively under inference mode. Tensors output of inference mode cannot be used for anything unless they are cloned.